hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/CDCFileGroupIterator.scala

```diff
@@ -112,7 +112,7 @@ class CDCFileGroupIterator(split: HoodieCDCFileGroupSplit,
       metadataConfig,
       // TODO support CDC with spark record
       recordMergerImpls = List(classOf[HoodieAvroRecordMerger].getName),
-      recordMergerStrategy = HoodieRecordMerger.DEFAULT_MERGER_STRATEGY_UUID
+      recordMergerStrategy = HoodieRecordMerger.PAYLOAD_BASED_MERGER_STRATEGY_UUID
```

Comment:
Looks like there are places that the config passing can be improved later.

Reply:
CDC doesn't support recordmerger, not on the read or write side

Reply:
Let's add a Jira ticket to track this.

Reply:
HUDI-8423 as a follow-up.